### DISCUSSION TITLE
Have a doubt on the code in Lecture 7 in DeployFundMe contract

### DISCUSSION THREAD

**Discussion by karthiksathishjeemain** [Cyfrin/foundry-full-course-cu#990] [Q&A] [answered]
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.19;

import {Script} from "forge-std/Script.sol";
import {HelperConfig} from "./HelperConfig.s.sol";
import {FundMe} from "../src/FundMe.sol";

contract DeployFundMe is Script {
    function run() external returns (FundMe, HelperConfig) {
        HelperConfig helperConfig = new HelperConfig(); // This comes with our mocks!
        address priceFeed = helperConfig.activeNetworkConfig();

        vm.startBroadcast();
        FundMe fundMe = new FundMe(priceFeed);
        vm.stopBroadcast();
        return (fundMe, helperConfig);
    }
}
In this code snippet, to whom( which particular keyword) does the function run  returns fundMe and helperConfig at the end?

**Comment by 0xusmanf** [comment, marked as answer]
Whoever will call this `run()` function will get `(fundMe, helperConfig)` instances in return. For example in the test file below these two instances will be saved like this `(s_fundMe, s_helperConfig) = deployFundMe.run();`  when calling the `run()` function.
```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Test} from "forge-std/Test.sol";
import {FundMe} from "../../src/FundMe.sol";
import {DeployFundMe} from "../../script/DeployFundMe.s.sol";
import {HelperConfig} from "../../script/HelperConfig.s.sol";

contract TestFundMe is Test {
    FundMe s_fundMe;
    HelperConfig s_helperConfig;

    function setUp() external {
        DeployFundMe deploy = new DeployFundMe();
        (s_fundMe, s_helperConfig) = deploy.run();
    }
}
```
